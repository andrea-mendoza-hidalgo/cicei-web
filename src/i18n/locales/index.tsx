import en from './translations.en.json';
import es from './translations.es.json';
import { SharedEn, SharedEs } from '../../shared/components/locale';
import { PagesEn, PagesEs } from '../../components/pages/locales';
import ProjectEn from '../../components/project/locale/project.en.json';
import ProjectEs from '../../components/project/locale/project.es.json';
import PublicationsEn from '../../components/publication/locale/publication.en.json';
import PublicationsEs from '../../components/publication/locale/publication.es.json';
import ResearchersEn from '../../components/researchers/locale/researchers.en.json';
import ResearchersEs from '../../components/researchers/locale/researchers.es.json';
import NewsEs from '../../components/news/locale/news.es.json';
import NewsEn from '../../components/news/locale/news.en.json';

en.shared= SharedEn;
es.shared= SharedEs;
en.pages = PagesEn;
es.pages = PagesEs;
en.project = ProjectEn;
es.project = ProjectEs;
en.publication = PublicationsEn;
es.publication = PublicationsEs;
en.researchers = ResearchersEn;
es.researchers = ResearchersEs;
en.news = NewsEn;
es.news = NewsEs;



export {en, es};