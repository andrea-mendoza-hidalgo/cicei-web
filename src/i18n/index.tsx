import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import { es, en } from './locales';

const options = {
  debug: false,

  resources: {
    'es-ES': {
      shared: es.shared,
      pages: es.pages,
      project: es.project,
      publication: es.publication,
      researchers: es.researchers,
      news: es.news
    },
    'en-US': {
      shared: en.shared,
      pages: en.pages,
      project: en.project,
      publication: en.publication,
      researchers: en.researchers,
      news: en.news
    }
  },

  fallbackLng: 'es-ES',

  ns: ['shared', 'pages', 'project', 'publication'],

  defaultNS: 'shared',

  react: {
    wait: true
  }
};

i18n.use(initReactI18next).use(LanguageDetector).init(options);

export default i18n;
