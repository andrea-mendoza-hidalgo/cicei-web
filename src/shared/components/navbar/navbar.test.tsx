import React from 'react';
import { mount } from 'enzyme';
import Navbar from './navbar';
import { Link, MemoryRouter } from 'react-router-dom';
import NavbarOptions from './navbar-options';
import { fireEvent, render, within } from '@testing-library/react';
import '../../../i18n';

describe('Navbar Render Test', () => {
  let navbar: any;
  beforeEach(async () => {
    navbar = mount(
      <MemoryRouter>
        <Navbar activeOption={'research'} activeDropdownOption={'projects'}/>
      </MemoryRouter>
    );
  });

  afterEach(() => {
    navbar.unmount();
  });
  it('Navbar renders correctly ', () => {
    expect(navbar).toMatchSnapshot();
  });

  it('when screen size it is lower than 960a and mouse enter, dropdown has to showed', async () => {
    global.innerWidth = 600;
    global.dispatchEvent(new Event('resize'));

    navbar.find('.menu__icon').simulate('click').simulate('click');

    const menu = navbar.find('.navbar__item').at(2).simulate('mouseenter');

    expect(menu.find(Link).props().to).toBe('/research');
  });

  it('when screen size it is lower than 960a and mouse leave, dropdown has to hide', async () => {
    global.innerWidth = 600;
    global.dispatchEvent(new Event('resize'));

    navbar.find('.menu__icon').simulate('click').simulate('click');

    navbar.find('.navbar__item').at(2).simulate('mouseenter');
    const menu = navbar.find('.navbar__item').at(2).simulate('mouseleave');

    expect(menu.find(Link).props().to).toBe('/research');
  });

  it('when screen size it is greater than 1020 and mouse enter, dropdown has to hide', async () => {

    global.innerWidth = 1020;
    global.dispatchEvent(new Event('resize'));

    navbar.find('.menu__icon').simulate('click').simulate('click');

    const menu = navbar.find('.navbar__item').at(2).simulate('mouseenter');

    expect(menu.find(Link).props().to).toBe('/research');
  });

  it('when screen size it is lower than 1020 and mouse leave, dropdown has to hide', async () => {
    global.innerWidth = 1020;
    global.dispatchEvent(new Event('resize'));

    navbar.find('.menu__icon').simulate('click').simulate('click');

    navbar.find('.navbar__item').at(2).simulate('mouseenter');
    const menu = navbar.find('.navbar__item').at(2).simulate('mouseleave');

    expect(menu.find(Link).at(0).props().to).toBe('/research');
  });
});

describe('Navbar options render text', () => {
  it('Navbar options renders correctly ', () => {
    const { getByRole } = render(
      <NavbarOptions
        languages={[
          { value: 'en-US', label: 'Eng' },
          { value: 'es-ES', label: 'Esp' }
        ]}
        currentLanguage={'en-US'}
      />
    );

    fireEvent.mouseDown(getByRole('button'));
    const listbox = within(getByRole('listbox'));
    fireEvent.click(listbox.getByText('Esp'));

    expect(listbox.getByText('Esp')).toBeInTheDocument();
  });
})