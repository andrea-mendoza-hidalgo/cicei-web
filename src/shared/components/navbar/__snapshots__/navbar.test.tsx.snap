// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Render Test Navbar renders correctly  1`] = `ReactWrapper {}`;
